checker/src/call_graph.rs

```diff
@@ -736,15 +859,27 @@ impl CallGraph {
                 Ok(relations) => relations,
                 Err(e) => panic!("Failed to read input type relations: {:?}", e),
             };
-            let mut input_type_relations = HashSet::<TypeRelation>::new();
-            for relation in input_type_relations_raw.relations.iter() {
-                input_type_relations.insert(relation.to_owned());
+            let input_relations = input_type_relations_raw.relations;
+            for relation in input_relations.iter() {
+                type_relations.insert(relation.to_owned());
+                match type_to_index.get(&relation.type1) {
+                    Some(_) => {}
+                    None => {
+                        max_id += 1;
+                        type_map.insert(max_id, relation.type1.to_owned());
+                    }
+                }
+                match type_to_index.get(&relation.type2) {
+                    Some(_) => {}
+                    None => {
+                        max_id += 1;
+                        type_map.insert(max_id + 2, relation.type2.to_owned());
```

Comment:
I find myself wondering why you add 2 to max_id here.

Reply:
That's a bug that was introduced through refactoring; thanks for pointing that out!